evm/vm/base.py

```diff
@@ -266,6 +275,8 @@ def validate_block(self, block):
                     )
                 )
 
+        # FIXME: make_trie_root_and_nodes() is rather expensive, and we already run that once in
+        # import_block(), so should refactor some of this code to avoid re-generating it here.
```

Comment:
This seems like we could fix this with some caching on the `make_trie_root_and_nodes` function (or maybe a wrapper around that function which implements a cache).

Reply:
Yeah, I guess #573 might still be useful!